plugin-scalebar/src/main/java/com/mapbox/pluginscalebar/ScaleBar.java

```diff
@@ -0,0 +1,91 @@
+package com.mapbox.pluginscalebar;
+
+import android.support.annotation.ColorRes;
+import android.support.annotation.NonNull;
+import android.support.annotation.UiThread;
+import android.support.annotation.VisibleForTesting;
+import android.view.View;
+
+import com.mapbox.mapboxsdk.camera.CameraPosition;
+import com.mapbox.mapboxsdk.maps.MapView;
+import com.mapbox.mapboxsdk.maps.MapboxMap;
+import com.mapbox.mapboxsdk.maps.Projection;
+
+/**
+ * Plugin class that shows a scale bar on MapView and changes the scale corresponding to the MapView's scale.
+ */
+public class ScaleBar {
+  private MapboxMap mapboxMap;
+  private Projection projection;
+  private boolean enabled = true;
+  private ScaleBarWidget scaleBarWidget;
+  private CameraMoveListener cameraMoveListener;
+
+  public ScaleBar(@NonNull MapView mapView, @NonNull MapboxMap mapboxMap) {
+    this.mapboxMap = mapboxMap;
+    this.projection = mapboxMap.getProjection();
+    this.scaleBarWidget = new ScaleBarWidget(mapView.getContext());
+    mapView.addView(scaleBarWidget);
+    cameraMoveListener = new CameraMoveListener();
+    mapboxMap.addOnCameraMoveListener(cameraMoveListener);
+    scaleBarWidget.setMapViewWidth(mapView.getWidth());
+  }
```

Comment:
The logic from `onCameraMove` will need to be invoked here to query the meters per pixel and invalidate the view. Otherwise, on an initial start, the bar is not going to show up, because the camera has not moved yet.